contrib/babelfishpg_tsql/src/pl_exec-2.c

```diff
@@ -236,6 +242,126 @@ exec_tsql_stmt(PLtsql_execstate *estate, PLtsql_stmt *stmt, PLtsql_stmt *save_es
 	return rc;
 }
 
+static int
+exec_stmt_kill(PLtsql_execstate *estate, PLtsql_stmt_kill *stmt)
+{
+	PGPROC *proc;    	
+	Oid	    sysadmin_oid = get_role_oid("sysadmin", false);  // should really use BABELFISH_SYSADMIN in tds_int.h
+	int	    spid = -1;
+	char   *strtol_endptr;
+
+	if (pltsql_explain_only)
+	{
+		StringInfoData query;
+
+		initStringInfo(&query);
+		appendStringInfo(&query, "KILL ");
+		appendStringInfoString(&query, stmt->spid);
+		append_explain_info(NULL, query.data);
+		return PLTSQL_RC_OK;
```

Comment:
We might want to pfree(query.data) after use

Reply:
done